### PR TITLE
Release auto mixed-precision training for bert

### PR DESCRIPTION
**Pull request by kuke**
Update BERT for the fp16 support.

### CODE REVIEW

File: PaddleNLP/language_representations_kit/BERT/optimization.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 import numpy as np
 import paddle.fluid as fluid
-from utils.fp16 import create_master_params_grads, master_param_to_train_param
+from utils.fp16 import create_master_params_grads, master_param_to_train_param, apply_dynamic_loss_scaling
```
Comment (gongweibao) [inline PaddleNLP/language_representations_kit/BERT/optimization.py:22]: Use decorate.

File: PaddleNLP/language_representations_kit/BERT/utils/fp16.py [hunk 2]
```diff
@@ -17,26 +17,20 @@
 import paddle.fluid as fluid
 
 
-def cast_fp16_to_fp32(i, o, prog):
+def append_cast_op(i, o, prog):
```
Comment (gongweibao) [inline PaddleNLP/language_representations_kit/BERT/utils/fp16.py:20]: These files should not exists.

**Review by gongweibao — CHANGES_REQUESTED** [review]

**Review by iclementine — APPROVED** [review]
Looks Good to Me!

File: PaddleNLP/language_representations_kit/BERT/optimization.py (hunk 1 above)
Comment (kuke) [reply PaddleNLP/language_representations_kit/BERT/optimization.py:22]: The decorate interface should be refined, before that we use the customized code here.

**Pull request merged**